### DISCUSSION TITLE
Connection timeout while first startup

### DISCUSSION THREAD

**Discussion by Nordiii** [photoprism/photoprism#2261] [Q&A] [answered]
Hello,

I'm trying to run this on my raspberry pi 4, I'm currently at around 40gb of Pictures. 

Starting up the container it works for a pretty long time (around half an hour I would say) and then fails.

This are the logs:
```bash
init as root
image: arm64-prod, build 220302-arm64
home: /photoprism
umask: "0002" (u=rwx,g=rwx,o=rx)
install-path: /opt/photoprism
running as uid 0
/opt/photoprism/bin/photoprism start
time="2022-04-18T06:23:46Z" level=info msg="PhotoPrism® needs your support!"
time="2022-04-18T06:23:46Z" level=info msg="Visit https://docs.photoprism.app/funding/ to learn more."
time="2022-04-18T06:23:46Z" level=info msg="config: make sure your server has enough swap configured to prevent restarts when there are memory usage spikes"
time="2022-04-18T06:52:58Z" level=fatal msg="dial tcp 172.18.0.2:3306: connect: connection timed out"
```

Im using  mysql/mysql-server as database container. 
First I changed the `interactive_timeout` and `wait_timeout` to `28800` which sadly still ends in the connection time out.

Is there anything else I would need to configure? Here are my current timeout settings when executing `SHOW VARIABLES LIKE '%timeout';`

Variable_name | Value |  
-- | -- | --
connect_timeout | 10
delayed_insert_timeout | 300
have_statement_timeout | YES
innodb_flush_log_at_timeout | 1
innodb_lock_wait_timeout | 50
innodb_rollback_on_timeout | OFF
interactive_timeout | 28800
lock_wait_timeout | 31536000
mysqlx_connect_timeout | 30
mysqlx_idle_worker_thread_timeout | 60
mysqlx_interactive_timeout | 28800
mysqlx_port_open_timeout | 0
mysqlx_read_timeout | 30
mysqlx_wait_timeout | 28800
mysqlx_write_timeout | 60
net_read_timeout | 30
net_write_timeout | 60
replica_net_timeout | 60
rpl_stop_replica_timeout | 31536000
rpl_stop_slave_timeout | 31536000
slave_net_timeout | 60
wait_timeout | 28800

</body>
</html>

The docker-compose for the prism container is basicly the default one except for the passwords.

Any help is much appricated!

**Comment by Nordiii** [comment, marked as answer]
Ok so, I did miss setting the network!

**Reply by osarusan82** [reply]
Can you explain what you mean by you "missed setting the network?"

I have the same issue. PhotoPrism worked for me for about 2 months, and then one day it stopped working and I have this exact same error message as you did, but I can't figure out what needs to be done to fix it.

**Reply by Nordiii** [reply]
Its now a while ago but if I remember correctly literally I forgot to set them into the same network/didn't configure it in my docker compose.
Therefore I had issues with connecting to the mysql container.

**Reply by osarusan82** [reply]
I see, it must have been a while ago since they stopped using mysql. They use mariadb now, and there doesn't seem to be any place to configure the network for that. I guess my problem lies elsewhere.

**Reply by Nordiii** [reply]
Sorry could totally be Mariadb instead of mysql. Use the later more often. The end problem was that I had the DB and prism container not in the same network.

**Reply by osarusan82** [reply]
Ah ok, so definitely not the same issue here. My db and container are both on the same PC.
